lib/resources/resourceBase.js

```diff
@@ -66,6 +66,13 @@ var ResourceBase = function (endpoint, config) {
           return reject(body.error);
         }
 
+        if (resp && resp.statusCode >= 500) {
+          return reject({
```

Comment:
I think this should be an `Error` object. Something like `new Error(resp.statusMessage + ' (' + resp.statusCode + ')')`